### PR TITLE
FIX Restore filters in the file search

### PR DESCRIPTION
**Pull request by maxime-rainville**
I did a booboo a wee while ago that breaks the filters. This undoes said booboo and adds some behat test to detect future failures.

Fixes https://github.com/silverstripe/silverstripe-asset-admin/issues/1158

### CODE REVIEW

File: client/src/containers/AssetAdmin/AssetAdmin.js [hunk 1]
```diff
@@ -675,28 +675,26 @@ class AssetAdmin extends Component {
   }
 
   render() {
-    const { folder, folderId, query, getUrl, type, maxFiles, filter, toolbarChildren } = this.props;
+    const { folder, folderId, query, getUrl, type, maxFiles, toolbarChildren } = this.props;
```
Comment (maxime-rainville) [inline client/src/containers/AssetAdmin/AssetAdmin.js:678]: All these props used to be access with the `this.props` syntax. When I did the in-modal editing job, I updated the syntax to use de-structuring instead. However, I screwed up the filter bit.

File: client/src/containers/AssetAdmin/AssetAdmin.js [hunk 2]
```diff
@@ -675,28 +675,26 @@ class AssetAdmin extends Component {
...
     );
-    const showSearch = hasFilters(filter) || this.props.showSearch;
+    const showSearch = hasFilters(query.filter) || this.props.showSearch;
     const onSearchToggle = this.props.actions.displaySearch ?
       this.props.actions.displaySearch.toggleSearch :
       undefined;
 
     const breadcrumbProps = {
       folder,
-      query: (this.props.query && this.props.query.view) ? Object.assign({
```
Comment (maxime-rainville) [inline client/src/containers/AssetAdmin/AssetAdmin.js:697]: This is an elaborate way of doing nothing.

File: tests/behat/features/manage-files.feature [hunk 3]
```diff
@@ -204,3 +204,22 @@ Feature: Manage files
     Then I should see the file named "file1" in the gallery
       And I should not see the file named "folder2" in the gallery
       And I should not see "File cannot be found" in the "#Form_fileEditForm" element
+
+  @search
```
Comment (maxime-rainville) [inline tests/behat/features/manage-files.feature:208]:
```suggestion
```

**Review by Cheddam — APPROVED** [review]
Happy with this code-wise, needs a manual stress test though.

**Pull request merged**